thorlcr/msort/tsortm.cpp

```diff
@@ -629,32 +624,26 @@ class CSortMaster: public IThorSorterMaster, public CSimpleInterface
                 CSortNode &slave = slaves.item(i);
                 unsigned slavesamples = averagerecspernode?((unsigned)((averagerecspernode/2+averagesamples*slave.numrecs)/averagerecspernode)):1;
                 //PrintLog("%d samples for %d",slavesamples,i);
-                if (slavesamples) {
+                if (slavesamples)
+                {
                     size32_t samplebufsize;
                     void *samplebuf=NULL;
                     slave.GetMultiNthRow(slavesamples, samplebufsize, samplebuf);
-                    CriticalBlock block(asect);
-                    sample.deserializeExpand(samplebufsize, samplebuf);
+                    MemoryBuffer mb;
+                    fastLZDecompressToBuffer(mb, samplebuf);
                     free(samplebuf);
+                    CriticalBlock block(asect);
```

Comment:
yes - it's because the rowarray has it's own serialization format (header and potentially markers for nulls) and now the slave is serializing each row through a rowToKey serializer into a buffer and compressing, i.e. they are not/can not be placed into a row array and then serialized.